### PR TITLE
Fixed bug preventing docker client to retrieve container logs when docker server has version prior to 1.25

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Prevent the docker client to use an API version when retrieving the docker API version of the server

### Motivation

Enabled users to use docker API engine with a version prior to `1.25`

### Additional notes

This bug fix this [issue](https://github.com/DataDog/datadog-agent/issues/1300).
It has been tested with docker API version `1.25` and `1.24`

### CODE REVIEW

File: pkg/logs/input/container/scanner.go [hunk 1]
```diff
@@ -226,6 +226,8 @@ func (s *Scanner) newDockerClient() (*client.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	// assure the client can communicate with the server
+	client.UpdateClientVersion("")
 	v, err := client.ServerVersion(context.Background())
```
Comment (tmichelet) [inline pkg/logs/input/container/scanner.go]: could we extract lines 230 and 231 into a method to ensure we always call `UpdateClientVersion("")` whenever we compute the server version?
also, could this benefit other parts of the agent? Would it be worth moving to common logic?

Comment (xvello) [reply pkg/logs/input/container/scanner.go]: FWI this is the logic infra+process agents use: https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/docker_util.go#L106-L132

This fix is cherry-picked from our 1.12 compat fixes.

Comment (ajacquemot) [reply pkg/logs/input/container/scanner.go]: I suggest keep using this piece of code and move to `docker_util` at some point, WDYT ?

**Comment by xvello** [comment]
Tested on docker 1.12.6:
```
2018-02-20 14:12:58 UTC | INFO | (logs.go:29 in Start) | Starting logs-agent
2018-02-20 14:12:58 UTC | INFO | (tagger.go:77 in Init) | starting the tagging system
2018-02-20 14:12:59 UTC | INFO | (tagger.go:151 in tryCollectors) | docker tag collector successfully started
2018-02-20 14:12:59 UTC | INFO | (scanner.go:257 in setupTailer) | Detected container datadog/agent-dev:ajacquemot-logs-docker-api - 72e3a958dfc0
2018-02-20 14:12:59 UTC | INFO | (scanner.go:257 in setupTailer) | Detected container xvellodatadog/agent-dev:universe - 64162fad93af
2018-02-20 14:12:59 UTC | INFO | (connection_manager.go:58 in NewConnection) | Connecting to the backend: intake.logs.datadoghq.com:10516
```

**Review by tmichelet — DISMISSED** [review]

**Comment by olivielpeau** [comment]
Let's add a release note on this PR

File: releasenotes/notes/logs-fix-docker-issue-api-version-1ed6a65d6c4322c6.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+issues:
```
Comment (olivielpeau) [inline releasenotes/notes/logs-fix-docker-issue-api-version-1ed6a65d6c4322c6.yaml]: this should be under `fixes`, not `issues`

**Review by olivielpeau — CHANGES_REQUESTED** [review]

**Review by olivielpeau — APPROVED** [review]
AppVeyor failure is unrelated. Merging

**Pull request merged**